File: Project_Sourcecode_go/state/Suspended.go
package state

type Suspended struct {
	State
}

func (s *Suspended) Close() {
	s.M.ChangeState(8)
}

func (s *Suspended) Balance() {
	s.M.P.DisplayBalance()
}

func (s *Suspended) Activate() {
	s.M.ChangeState(3)
}
func (s *Suspended) GetName() string {
	return "Suspended"
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/state/Suspended.java
package edu.iit.cs586.project.state;

import edu.iit.cs586.project.MDA_EFSM;
import edu.iit.cs586.project.output.OP;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 22:33
 */
public class Suspended extends State{
    public Suspended(MDA_EFSM m) {
        super(m);
    }

    @Override
    public String getName() {
        return "Suspended";
    }

    @Override
    public void activate() {
        m.changeState(3);
    }

    @Override
    public void close() {
        m.changeState(8);
    }

    @Override
    public void balance() {
        m.p.displayBalance();
    }
}
